modules/core/src/main/scala/fs2/kafka/internal/package.scala

```diff
@@ -0,0 +1,15 @@
+package fs2.kafka
+
+import cats.effect.{Blocker, ContextShift, Resource, Sync}
+
+package object internal {
+  type ExclusiveAccess[F[_], A] = F[A] => F[A]
```

Comment:
Rather than add another abstraction for exclusive access I made a type alias. This could cause type inference problems in the future as it's not partially applied, but I think it's probably fine...

Reply:
As long as it's all package-private let's go with it - we can always refactor it later on.